app/components/TransactionHistory/TransactionHistoryPanel/TransactionHistoryPanel.jsx

```diff
@@ -24,11 +30,39 @@ export default class TransactionHistory extends React.Component<Props> {
         <Transactions
           className={styles.transactions}
           transactions={transactions}
+          pendingTransactions={pendingTransactions || []}
         />
       </Panel>
     )
   }
 
+  async pruneReturnedTransactionsFromStorage() {
+    const {
+      transactions,
+      pendingTransactions,
+      address,
+      net,
+      getPendingTransactionInfo,
+    } = this.props
+    if (
+      transactions &&
+      transactions.length &&
+      pendingTransactions &&
+      pendingTransactions.length
```

Comment:
nit: Not sure this check is actually needed.
`transactions` and `pendingTransactions` will always be arrays (or Flow would have raised an error) and if empty the following code would still work fine.